packages/x-date-pickers/src/internals/hooks/useDesktopPicker/useDesktopPicker.tsx

```diff
@@ -138,14 +124,24 @@ export const useDesktopPicker = <
       sx,
       format,
       formatDensity,
+      enableAccessibleFieldDOMStructure,
+      selectedSections,
+      onSelectedSectionsChange,
       timezone,
       label,
       name,
       autoFocus: autoFocus && !props.open,
       focused: open ? true : undefined,
+      ...(inputRef ? { inputRef } : {}),
     },
     ownerState: props,
-  });
+  }) as BaseSingleInputFieldProps<
```

Comment:
I had a look at it and it seems that the reason why we have to do `as` casting is the fact that we have a dynamic `PickerValidDate` type that does not work in our monorepo. 🙈 
Am I correct? 🤔

Reply:
I had the casting before the `PickerValidDate` PR was merged for what I remember
But maybe it has an effect here :thinking:

Reply:
From what I understand, the value returned by `useSlotProps` uses `PickerValidDate` instead of `TDate` (at least for `referenceDate`)
But it's strange, all the elements returned are using `TDate` and `referenceDate` does not even exist on the returned value...

Reply:
Looks like I can keep the generics on `useSlotProps` like before, I will just simplify it a bit because the super long list of `Pick` is not great

Reply:
Done :+1: